docs/fuel-101/transaction-lifecycle.mdx

```diff
@@ -5,4 +5,54 @@ category: Fuel 101
 
 # Transaction Lifecycle
 
-Fuel's transaction lifecycle explained.
+A transaction on the Fuel network goes through a complex series of steps before it is considered final:
+
+1. **Transaction Creation**: Someone sends a transaction to a node.
+2. **Pre-Validation**: The transaction goes through an extensive series of pre-checks, including UTXO validation.
+3. **Transaction Pool**: The transaction gets placed in the transaction pool.
+4. **Selection**: Transactions are selected from the pool.
+5. **Execution & Validation**: The transaction is executed in the FuelVM.
+6. **Block Production**: The transaction is included in a block.
+7. **Gossiping & Validation**: The block is broadcast to the network and the validatation process is repeated.
+8. **Finality on Ethereum**: The transaction is considered final.
+
+## 1. Transaction Creation
+
+The transaction is created by someone.
+
+One way a transaction can be created is by interacting with an application that uses the `fuels` SDK. In this case, the SDK will do some validation to make sure that the transaction appears valid before sending it to the node. If the SDK detects that the transaction is not valid, the transaction will not move on to the next step.
+
+## 2. Pre-Validation
+
+All transactions are fully [validated](/docs/fuel-101/transaction-model#transaction-validation) by the Fuel VM.
+
+Before a transaction makes it to the transaction pool, it goes through a extensive pre-checks. These checks are done to ensure that the transaction appears valid before attempting to execute it.
+
+## 3. Transaction Pool
+
+The transaction pool sorts pending transactions by their gas price. Note that a transaction's gas represents the total amount of work that the VM did to execute the transaction, while the gas price represents the amount that the transaction is going to pay for each unit of gas (or `GasSpend`).
+
+## 4. Selection
+
+Transactions are selected from the pool to be included in the block. They are selected in a way that maximizes the block producer's reward.
+
+## 5. Execution & Validation
+
+If a transaction contains a script, the script will be executed.
+The transaction execution is completed by removing the spent UTXOs and adding any new UTXOs that were created.
+
+The transaction undergoes further validation at this step.
+If the transaction contains a predicate, the predicate is executed and validated.
+
+## 6. Block Production
+
+The block producer creates a new block that includes the transaction.
+
+## 7. Gossiping & Validation
+
+The block is broadcast to the network, and the nodes gossip with each other.
+Validators connected to the block producer will receive the block, repeat the validation and execution steps, and continue to broadcast the block.
+
+## 8. Finality on Ethereum
+
+(is this happening now?)
```

Comment:
I don't think this is happening now in beta-5, but this section could be in the future tense for now and updated later to reflect present tense reality when it changes.

Reply:
@luizstacio does the block committer for the bridge essentially do this for all transactions?

Reply:
right now the block committer only finalizes withdrawals via state root commitments. Transactions finality on DA won't occur until we complete full DA integration.